core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -161,6 +162,18 @@ sealed abstract class IO[+E, +A] extends Serializable { self =>
         )
     )
 
+  /**
+   * Races this action with the specified action, returning the first
+   * result to *finish*, whether with by producing a value or by failing
+   * with an error. If either of two actions fails before the other succeeds,
+   * the whole race will fail with that error.
+   */
+  final def raceAttempt[E1 >: E, B](that: IO[E1, B]): IO[E1, Either[A, B]] =
```

Comment:
Naming to discuss in #311 perhaps, as it is the only `race` op that fails early.

Reply:
You can `.merge` an Either to recover an `A` though, seems like chicken-and-egg to me. However, as `.race` unifies by default, might as well do that here as well.